examples/dynamic-bridge-app/linux/main.cpp

```diff
@@ -176,6 +189,141 @@ std::unique_ptr<GeneratedCluster> CreateCluster(chip::ClusterId id)
     return nullptr;
 }
 
+void ReadValueFromBuffer(chip::TLV::TLVWriter & wr, const char * buffer, const Span<const char> *)
+{
+    wr.PutString(chip::TLV::Tag(), buffer, (uint32_t) strlen(buffer));
+}
+void ReadValueFromBuffer(chip::TLV::TLVWriter & wr, const char * buffer, const ByteSpan *)
+{
+    wr.PutBytes(chip::TLV::Tag(), (const uint8_t *) buffer, (uint32_t) strlen(buffer));
+}
+void ReadValueFromBuffer(chip::TLV::TLVWriter & wr, const char * buffer, const float *)
+{
+    wr.Put(chip::TLV::Tag(), (float) atof(buffer));
+}
+void ReadValueFromBuffer(chip::TLV::TLVWriter & wr, const char * buffer, const double *)
+{
+    wr.Put(chip::TLV::Tag(), atof(buffer));
+}
+void ReadValueFromBuffer(chip::TLV::TLVWriter & wr, const char * buffer, const int64_t *)
+{
+    wr.Put(chip::TLV::Tag(), (int64_t) strtoll(buffer, nullptr, 10));
+}
+void ReadValueFromBuffer(chip::TLV::TLVWriter & wr, const char * buffer, const uint64_t *)
+{
+    wr.Put(chip::TLV::Tag(), (uint64_t) strtoll(buffer, nullptr, 10));
+}
+void ReadValueFromBuffer(chip::TLV::TLVWriter & wr, const char * buffer, const bool *)
+{
+    wr.PutBoolean(chip::TLV::Tag(), (*buffer) ? true : false);
+}
+
+template <typename T>
+void ReadValueFromBuffer(chip::TLV::TLVWriter & wr, const char * buffer, const T *)
+{
+    wr.Put(chip::TLV::Tag(), (int64_t) strtoll(buffer, nullptr, 10));
+}
+
+void WriteValueToBuffer(const bool & value, uint8_t * buffer, const uint16_t & maxReadLength)
+{
+    if (maxReadLength == 1)
+        *buffer = value ? 1 : 0;
+}
+
+template <typename T>
+void WriteValueToBuffer(const T & value, uint8_t * buffer, const uint16_t & maxReadLength)
+{
+    size_t value_size = sizeof(value);
+    if (maxReadLength >= value_size)
+    {
+        memcpy(buffer, &value, value_size);
+    }
+}
+
+template <typename T>
+void WriteValueToBuffer(chip::TLV::TLVReader & reader, uint8_t * buffer, const uint16_t & maxReadLength)
+{
+    T v;
+    chip::app::DataModel::Decode(reader, v);
+    WriteValueToBuffer(v, buffer, maxReadLength);
+}
+
+// if `read` is true, read from `buffer`, and write to `data`
+// otherwise, read from `data`, and write to `buffer`
+template <typename T>
+void ReadOrWriteBuffer(std::vector<uint8_t> * data, uint8_t * buffer, uint16_t maxReadLength, bool read)
+{
+    if (read)
+    {
+        chip::TLV::TLVWriter wr;
+        wr.Init(data->data(), data->size());
+        ReadValueFromBuffer(wr, (char *) buffer, (const T *) nullptr);
```

Comment:
use C++-style casts instead of C casts. Why do we assume buffer is a char * when it is given as binary?

Reply:
Addressed in [42661f6](https://github.com/project-chip/connectedhomeip/pull/23840/commits/42661f626408834ef130441ce658c9b021ff1e02)